mlrun/runtimes/nuclio/application/application.py

```diff
@@ -126,22 +128,68 @@ def __init__(
         self.min_replicas = min_replicas or 1
         self.max_replicas = max_replicas or 1
 
+        # initializing internal application port and application ports
+        self._internal_application_port = None
+        self._application_ports = []
+
+        application_ports = application_ports or []
```

Comment:
```suggestion
        self._application_ports = application_ports or []
```

Then no need for line 148

Reply:
@TomerShor there is a need for this. `application_ports` is an object with setter/getter, which should be run here. Init of private objects is a must as well as aplpication_ports and internal_application_port setters and getters are connected and we must ensure that both private objects exists before setting these methods